Doc/c-api/gcsupport.rst

```diff
@@ -228,3 +228,36 @@ garbage collection runs.
    Returns the current state, 0 for disabled and 1 for enabled.
 
    .. versionadded:: 3.10
+
+
+Querying Garbage Collector State
+--------------------------------
+
+The C-API provides the following inerface for querying information about
+the garbage collector.
+
+.. c:function:: void PyUnstable_GC_VisitObjects(gcvisitobjects_t callback, void *arg)
+
+   Run supplied callback on all live GC-capable objects.
+
+   .. warning::
+      If new objects are (de)allocated by the callback it is undefined if they
+      will be visited.
+
+   .. warning::
+      Garbage collection is disabled during operation. Explicitly running a collection
+      in the callback may lead to undefined behaviour e.g. visiting the same objects
+      multiple times or not at all.
+
+   .. versionadded:: 3.12
+
+.. c:type:: int (*gcvisitobjects_t)(PyObject *object, void *arg)
+
+   Type of the visitor function to be passed to :c:func:`PyUnstable_GC_VisitObjects`.
+   ``arg`` is the same as passed to ``PyUnstable_GC_VisitObjects``. Returning ``0``
+   allows iteration to continue, returning ``1`` stops further iteration. Returning any
+   other value has undefined behavior.
```

Comment:
- Fix argument format (pair of \*, not backticks)
- Slightly more succinct wording regarding return values

```suggestion
   *arg* is the same as the *arg* passed to ``PyUnstable_GC_VisitObjects``.
   Return ``0`` to continue iteration, return ``1`` to stop iteration.
   Returning any other value is undefined behavior.
```

Perhaps the last sentence should be a `.. warning::`?